### PR TITLE
Allow topics to be ordered by admins in the sidebar

### PR DESCRIPTION
**Pull request by gdjohnson**
Somewhat a WIP. Basic functionality in, could do with some QA, due diligence, code cleanup, etc. 

To test, use Admin dropdown caret, open Order Topics modal, drag-drop rows. On 'Save' should update backend and persevere.

---

Additionally:
- Rips out old featured_order/featured_tags functionality from chain meta (no longer used, naming became confusing)
- Adds some error checking and name uniqueness enforcement to the NewTopicModal.
- Reinstates our Dragula support

### CODE REVIEW

File: server/routes/orderTopics.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { DB } from '../database';
+
+enum OrderTopicsErrors {
+  NoUser = 'Not logged in',
+  NoIds = 'Must supply ordered array of topic IDs',
+  NoChain = 'Must supply a chain ID',
+  NoPermission = `You do not have permission to order topics`,
+}
+
+// TODO Graham 3/29/22: Add checks to ensure only featured tags are ordered
```
Comment (zakhap) [inline server/routes/orderTopics.ts:15]: ?

File: server/routes/updateChain.ts [hunk 2]
```diff
@@ -128,8 +131,9 @@ const updateChain = async (
   if (custom_stages) chain.custom_stages = custom_stages;
   if (terms) chain.terms = terms;
   if (snapshot) chain.snapshot = snapshot;
-  if (req.body['featured_topics[]'])
-    chain.featured_topics = req.body['featured_topics[]'];
+
+  // TODO Graham 3/31/22: Will this potentially lead to undesirable effects if toggle
```
Comment (zakhap) [inline server/routes/updateChain.ts:135]: ?

**Review by zakhap — CHANGES_REQUESTED** [review]
Works for me but you left a few TODOs, just wanna make sure things are A-OK for shipping

File: server/routes/orderTopics.ts (hunk 1 above)
Comment (gdjohnson) [reply server/routes/orderTopics.ts:15]: Eh yeah I'll add this in. In theory you'd have to be an admin hitting our routes (i.e. not going through the UI) and giving a ranking to feature tags, so it's unlikely, but I'll add it.

File: server/routes/updateChain.ts (hunk 2 above)
Comment (gdjohnson) [reply server/routes/updateChain.ts:135]: It's old code I'm worried about, so not relevant to this PR.

**Review by zakhap — APPROVED** [review]
LGTM 🚀

**Pull request merged**